Added import of TXT records into Custom zone file entries

svamberg opened the pull request:
The import-zone command write the _dmarc record into Custom zone file entries.

svamberg committed:
Added import of _dmarc records

The import-zone command write the _dmarc record into Custom zone file entries.

svamberg committed:
Added import of _dmarc records

The import-zone command write the _dmarc record into Custom zone file entries.

valtri commented:
Nice trick!

There could be added the check for `@{$rec->{TXT}} > 0`. Or even better, remove the condition for "_dmarc" hostname and use only this check for any standalone TXT record. (non-standalone TXT records should be covered by _hosttype_ 1)

We need to implement the full support for plain TXT records in the future, and this change would be compatible with it (just replacing the custom zone entries).

valtri commented on import-zone:
```diff
@@ -187,6 +187,16 @@ foreach $host (sort keys %zonedata) {
   $hosttype=8 if (@{$rec->{SRV}} > 0);
 
```
I would add something like this:

```
if (@{$rec->{TXT}} > 0) {
     $res = add_record("txt_entries", 
                        { type=>12,
                          ref=>$zoneid,
                          txt=>"$host2 $rec->{TTL} IN TXT \"$rec->{TXT}[0]\""
                        });
      fatal("failed to insert _dmarc record:\n   $host2 $rec->{TTL} IN TXT \"$rec->{TXT}[0]\"") if ($res < 0);
      next;
}
```

svamberg committed:
Generalization of TXT zone records import

TXT zone records (not having an A or AAAA record) are
inserted into Custom zone file entries.

svamberg committed:
Generalization of TXT zone records import

TXT zone records (not having an A or AAAA record) are
inserted into Custom zone file entries.

svamberg committed:
better code indentation

svamberg committed:
better code indentation

svamberg committed:
Fix comment

svamberg committed:
Fix comment

svamberg committed:
fix importing TXT records

svamberg committed:
fix importing TXT records

svamberg committed:
Merge branch 'fix-import-dmarc' of github.com:CESNET/sauron into fix-import-dmarc

svamberg committed:
Merge branch 'master' into fix-import-dmarc

svamberg committed:
Merge branch 'master' into fix-import-dmarc

svamberg committed:
Update ChangeLog

Removing forgotten characters

valtri reviewed (approved):
Looks good!
